src/navigation/context.py

```diff
@@ -121,6 +121,28 @@ def current_waypoint(self) -> Optional[mrover.msg.Waypoint]:
             return None
         return self.course_data.waypoints[self.waypoint_index]
 
+    def look_for_gate(self) -> bool:
+        """
+        Returns whether the currently active waypoint (if it exists) indicates
+        that we should go to a gate
+        """
+        waypoint = self.current_waypoint()
+        if waypoint is not None:
+            return waypoint.gate
+        else:
+            return False
+
+    def look_for_post(self) -> bool:
+        """
+        Returns whether the currently active waypoint (if it exists) indicates
+        that we should go to a post
+        """
+        waypoint = self.current_waypoint()
+        if waypoint is not None:
+            return (not waypoint.gate) and waypoint.post
```

Comment:
why is this one not just `return waypoint.post`

Reply:
I dont ever want look for gate and look for course to both be true because then the behavior of the rover would be dependent on the order of the if statements so we (somewhat arbitrarily admittedly) defined in the ICD that waypoint.post is only applicable if waypoint.gate is false and is otherwise ignored

Reply:
and yea I can build the safety into the calls